pkg/policyfilter/state.go

```diff
@@ -205,7 +205,12 @@ func (pol *policy) podMatches(podNs string, podLabels labels.Labels) bool {
 		return false
 	}
 
-	return pol.podSelector.Match(podLabels)
+	podLabels1 := podLabels
```

Comment:
There is a test failure, which seems to happen in this code because we are calling it with a `nil` podLabels. Can you PTAL?

```
 --- FAIL: TestK8s (0.67s)
    --- FAIL: TestK8s/containers_change (0.00s)
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 15 [running]:
testing.tRunner.func1.2({0x1886580, 0x1d630b0})
	/opt/hostedtoolcache/go/1.21.6/x64/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.21.6/x64/src/testing/testing.go:1548 +0x397
panic({0x1886580?, 0x1d630b0?})
	/opt/hostedtoolcache/go/1.21.6/x64/src/runtime/panic.go:914 +0x21f
github.com/cilium/tetragon/pkg/policyfilter.(*policy).podMatches(0xc0001999f8, {0x1aef0f5, 0x7}, 0x0?)
	/home/runner/work/tetragon/tetragon/go/src/github.com/cilium/tetragon/pkg/policyfilter/state.go:210 +0xf1
github.com/cilium/tetragon/pkg/policyfilter.(*policy).podInfoMatches(...)
	/home/runner/work/tetragon/tetragon/go/src/github.com/cilium/tetragon/pkg/policyfilter/state.go:217
github.com/cilium/tetragon/pkg/policyfilter.(*state).AddPolicy(0xc00011f5c0, 0x2, {0x0, 0x0}, 0xc000305040?)
	/home/runner/work/tetragon/tetragon/go/src/github.com/cilium/tetragon/pkg/policyfilter/state.go:408 +0x494
github.com/cilium/tetragon/pkg/policyfilter.testContainersChange(0xc000305040?, 0xc00011f5c0, 0xc00011f560?)
	/home/runner/work/tetragon/tetragon/go/src/github.com/cilium/tetragon/pkg/policyfilter/k8s_test.go:494 +0x176
github.com/cilium/tetragon/pkg/policyfilter.TestK8s.func6(0x0?)
	/home/runner/work/tetragon/tetragon/go/src/github.com/cilium/tetragon/pkg/policyfilter/k8s_test.go:605 +0x1b
testing.tRunner(0xc000305040, 0xc000111248)
	/opt/hostedtoolcache/go/1.21.6/x64/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 7
	/opt/hostedtoolcache/go/1.21.6/x64/src/testing/testing.go:1648 +0x3ad
FAIL	github.com/cilium/tetragon/pkg/policyfilter	0.684s
```